### PR TITLE
Tempo: Add kind to TraceQL intrinsics

### LINKED ISSUE #64905: [TraceQL] Support kind in Intellisense
Tempo is adding support for the intrinsic `kind` to TraceQL:
https://github.com/grafana/tempo/pull/2217

Auto suggestions work because they are pulled from the backend:

image

But Grafana has a hardcoded list of intrinsics here:
image

Can we add kind to this list? There may also be other changes I'm unaware of w/r to span intrinsics. Please review to see if `kind` support should be added anywhere else.

### PR DESCRIPTION
**Pull request by adrapereira**
**What is this feature?**

Small PR to add `kind` to the list of TraceQL intrinsics. Also refactored usages of **scopes** and **intrinsics** to link to a single place and avoid duplicated code. 

**Why do we need this feature?**

Tempo recently introduced support for `kind` in TraceQL queries and the frontend should be aligned https://github.com/grafana/tempo/pull/2217

**Who is this feature for?**

All users who wish to write TraceQL queries using a Tempo datasource. 

**Which issue(s) does this PR fix?**:

Fixes #64905

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.
- [ ] There are no known compatibility issues with older supported versions of Grafana, or plugins.
- [ ] It passes the [Hosted Grafana feature readiness review](https://docs.google.com/document/d/1QL9Ly8KnXzpb6ISbg49pTODRO5mhA5tkkfIZVX6pqQU/edit) for observability, scalability, performance, and security.

### CODE REVIEW

**Comment by aocenas** [comment]
I wonder would it be possible to get the intrinsics from server? cc @joe-elliott ? Right now it seems if I have a tempo version where kind isn't supported I get 
```
error querying ingesters in Querier.SearchTagValues: failed to get response from ingesters: failed to execute f() for 127.0.0.1:9095: rpc error: code = Unknown desc = tag name is not valid intrinsic or scoped attribute: kind
```
would be probably nicer if we know what intrinsics to show.

@adrapereira also on one hand here it's beneficial that we don't show that error, but seems like any error from the autocomplete is ignored, wonder if should be showing such error in error toast.

File: public/app/plugins/datasource/tempo/traceql/traceql.ts [hunk 1]
```diff
@@ -24,9 +24,9 @@ export const languageConfiguration = {
 
 export const operators = ['=', '!=', '>', '<', '>=', '<=', '=~'];
 
-const intrinsics = ['duration', 'name', 'status', 'parent'];
+export const intrinsics = ['duration', 'kind', 'name', 'status'];
```
[the remark this answers is not in the document]
Comment (joe-elliott) [reply public/app/plugins/datasource/tempo/traceql/traceql.ts:27]: parent is not a currently supported intrinsic. agree with removing it.

**Comment by joe-elliott** [comment]
> I wonder would it be possible to get the intrinsics from server? cc @joe-elliott ? Right now it seems if I have a tempo version where kind isn't supported I get

@aocenas I have been thinking about this lately and agree completely.

[This issue](https://github.com/grafana/tempo/issues/2233) was filed recently and I intend to fix it by introducing a `/api/v2/search/tags` for traceql. In designing this API I had considered:

- adding support for passing scopes. Something like `/api/v2/search/tags?scope=resource`
- return both intrinsics and tag names 

I will provide more details as I make progress.

**Comment by joe-elliott** [comment]
The PR is up to add scope to `/api/search/tags` and it includes an "intrinsic" scope which is a little cheaty, but works. It's blocked atm, but we should be able to get it in in the next few weeks.

https://github.com/grafana/tempo/pull/2282

**Review by joey-grafana — APPROVED** [review]

**Pull request merged**